lib/licensed/sources/gradle.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+require "tempfile"
+require "csv"
+require "uri"
+require "json"
+require "net/http"
+
+module Licensed
+  module Sources
+    class Gradle < Source
+      def self.type
+        "gradle"
+      end
+
+      def configurations
+        ["runtime", "runtimeClasspath"]
+      end
+
+      def initialize(config)
+        @config = config
+      end
+
+      def enabled?
+        (Licensed::Shell.tool_available?("gradle") || File.executable?(@config.pwd.join("gradlew"))) && File.exist?(@config.pwd.join("build.gradle"))
```

Comment:
could this be `Licensed::Shell.tool_available?("gradlew")` instead of checking for that executable file?  Why does this require that `gradlew` be in the current directory?

Reply:
No, this is deliberately `gradlew` in the project working directory. The standard approach for Gradle projects is that that script is then used to invoke Gradle for that project. As fallback it also allows `gradle` in the path, but most common projects include this file. Also see this example application: https://github.com/github/components-java/tree/master/components/rest-spring-boot